### DISCUSSION TITLE
How to handle legacy editorStates saved in database

### DISCUSSION THREAD

**Discussion by charklewis** [facebook/lexical#3586] [Q&A]
I have a number of legacy states saved in production databases that used the `convertLegacyJSONEditorState` function to render. Now that this has been removed, what is the best approach to handle these values?

The answer on this discussion will need to be updated too - https://github.com/facebook/lexical/discussions/2414.

**Comment by trueadm** [comment]
What I'd recommend doing, is staying on 0.6, and applying a script using headless mode that takes the old formats in the database and converts them to the new format and re-saves them back to the database. This is what a bunch of folks on Discord have mentioned to us as reliable methods of upgrading older structures. The good thing is that once you upgrade there shouldn't be any more breaking changes as the JSON structure is public – unlike the older structure where we weren't sure of how it would scale.

If that doesn't work, you could try forking the logic of `convertLegacyJSONEditorState` into your project and making that work for you.
